pkg/3scale/amp/operator/system_reconciler.go

```diff
@@ -248,12 +250,23 @@ func (r *SystemReconciler) validateS3StorageProvidedConfiguration() error {
 	var result *string
 	result = helper.GetSecretDataValue(secretData, component.AwsAccessKeyID)
 	if result == nil {
-		return fmt.Errorf("Secret field '%s' is required in secret '%s'", component.AwsAccessKeyID, awsCredentialsSecretName)
+		r.Logger().Info("AwsAccessKeyID was not found. Trying get STS fields")
+		result = helper.GetSecretDataValue(secretData, component.AwsRoleArn)
+		if result == nil {
+			return fmt.Errorf("Neither '%s' nor '%s' Secret field found in secret '%s'", component.AwsAccessKeyID, component.AwsRoleArn, awsCredentialsSecretName)
+		}
+		sts = true //if we are here - it's sts mode
+		result = helper.GetSecretDataValue(secretData, component.AwsSecretAccessKey)
```

Comment:
this does look right think we should be checking the AWS_WEB_IDENTITY_TOKEN_FILE here



<br class="Apple-interchange-newline">

Reply:
oh ... sorry .. didn't complete. Fixed. Thank you very much